content/en/tracing/trace_ingestion/mechanisms.md

```diff
@@ -0,0 +1,193 @@
+// set analytics_enabled by default
+tracerconfig.WithAnalytics(on bool)
+// set raw sampling rate to apply on all analytics_enabled spans
+tracerconfig.SetAnalyticsRate(0.4)
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+Tag any single span with `analytics_enabled:true`. In addition, specify a sampling rate to be associated with the span:
+```
+// in dd-trace-go
+// make a span analytics_enabled
+span.SetTag(ext.AnalyticsEvent, true)
+// make a span analytics_enabled with a rate of 0.5
+s := tracer.StartSpan("redis.cmd", AnalyticsRate(0.5))
+```
+
+### In the agent
+
+In the agent, an additional rate limiter is set to 200 spans per second. If the limit is reached, some spans are be dropped and not forwarded to Datadog.
+
+The rate can be set in the agent main configuration file (datadog.yaml) or as an environment variable :
+```
+@param max_events_per_second - integer - optional 200
+@env DD_APM_CONFIG_MAX_EVENTS_PER_SECOND - integer - optional 200
+```
+
+## Error and rare traces
+
+For traces not caught by the head-based sampling, agent mechanisms make sure that critical and diverse traces are kept and ingested. These 2 samplers keep a diverse set of traces (by catching all combinations of a predetermined set of tags):
+
+- **Error traces :** sampling errors is important to provide visibility on potential system failures
+- **Rare traces :** sampling rare traces allows you to keep visibility on your system as a whole, by making sure that low-traffic services and resource are still monitored.
+
+
+### Error traces
+_ingestion\_reason : error_
+
+The error sampler catches pieces of traces containing error spans that are not caught by head-based sampling. It distributes a `10 traces-per-second` rate to catch all combinations of `service`, `name`, `resource`, `http.status` and `error.type`.
+
+From agent version 7.33, the error sampler can be configured in the agent main configuration file (datadog.yaml) or with environment variables :
+```
+@param errors_per_second - integer - optional - default: 10
+@env DD_APM_ERROR_TPS - integer - optional - default: 10
+```
+
+**Note:** Set the parameter to `0` to disable the error sampler
+
+### Rare traces
+_ingestion\_reason : rare_
+
+The rare sampler sends a set of rare spans to Datadog.
+Rare sampler are also distributes a rate to catch combinations of `env`, `service`, `name`, `resource`, `error.type` and `http.status`. The default sampling rate for rare traces is `5 traces-per-second`.
+
+From agent version 7.33, the rare sampler can be disabled in the agent main configuration file (datadog.yaml) or with an environment variable :
+
+```
+@params apm_config.disable_rare_sample - boolean - optional - default: false
+@env DD_APM_DISABLE_RARE_SAMPLER - boolean - optional - default: false
+```
+
+**Note:** Sampled traces may be incomplete, as this mechanism occurs downstream of the head-based sampling. There is no way to guarantee that the agent will receive a complete trace from the tracing libraries.
```

Comment:
```suggestion
**Note**: Sampled rare traces may be incomplete, because this mechanism occurs downstream of the head-based sampling. It's possible that the Agent will receive an incomplete rare trace from the tracing libraries.
```